lib/ipmi_sdr.c

```diff
@@ -204,7 +204,9 @@ sdr_convert_sensor_reading(struct sdr_record_full_sensor *sensor, uint8_t val)
 	case 1:
 		if (val & 0x80)
 			val++;
-		/* Deliberately fall through to case 2. */
+		result = (double) (((m * (int8_t) val) +
+				    (b * pow(10, k1))) * pow(10, k2));
+		break;
```

Comment:
Why? The comment is totally valid from gcc's point of view and correctly silences the implicit fall through warning.
See https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-fallthrough

You here add unnecessary code duplication. There were no warnings in regard to this in ipmi_sdr.c. At least with gcc 5.4.0 on my Linux Mint 18.3 (Ubuntu Xenial). Wasn't it so on your system? What compiler?

Reply:
GCC removes the duplicate code and it gets rid of a warning. I am using gcc 7.3.0-27ubuntu1~18.04 on Ubuntu 18.04.

Reply:
The concern about duplicate code is not that it occupies space. It is that it is a source of bugs. It is not uncommon when someone changes the to-be-identical code in one place and then forgets to change it in the rest of the copies.

The existing comment indeed does not work. However, even gcc 8.1.0 is happy with a `/* fall through */` comment. So please just replace it with `/* fall through */`.

Reply:
I am all for code depuplication. I also hate warnings when code compiles. I am of a mind of either fix the warnings or what is the point of having the compiler warn you? Really what I want is a hammer to tell the compiler that I really know what I am doing in this case to make it shut up. It looks like /* fall through */ does just that. I was not aware of this trick.